adapters/tradplus/tradplus.go

```diff
@@ -0,0 +1,197 @@
+package tradplus
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"text/template"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/errortypes"
+	"github.com/prebid/prebid-server/v2/macros"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint *template.Template
+}
+
+// Builder builds a new instance of the tradplus adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	template, err := template.New("endpointTemplate").Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
+	}
+
+	return &adapter{
+		endpoint: template,
+	}, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	adapterRequest, errs := a.makeRequest(request)
+	if errs != nil {
+		return nil, errs
+	}
+	return []*adapters.RequestData{adapterRequest}, nil
+}
+
+func (a *adapter) makeRequest(request *openrtb2.BidRequest) (*adapters.RequestData, []error) {
+	errs := validateTradPlusExt(request)
+	if errs != nil {
+		return nil, errs
+	}
+
+	tradplusExt, err := getImpressionExt(&request.Imp[0])
```

Comment:
@tradplus Can you please confirm if you want to use ext of first impression even in case of multiple imps? If that is the case then `validateTradPlusExt()` function can be only used for first imp rather than looping over all the imps or validation for accountId can be done in `getImpressionExt()` function itself and return error from `getImpressionExt()`

Reply:
the `validateTradPlusExt` method has been deleted.